build/makelib/golang.mk

```diff
@@ -177,6 +183,12 @@ $(GO_TOOLS_DIR)/glide:
 	@mv $(GO_TOOLS_DIR)/$(GOHOSTOS)-$(GOHOSTARCH)/glide $(GO_TOOLS_DIR)/glide
 	@rm -r $(GO_TOOLS_DIR)/$(GOHOSTOS)-$(GOHOSTARCH)
 
+.PHONY: $(GO_TOOLS_DIR)/golint
+$(GO_TOOLS_DIR)/golint:
+	@echo "installing golint"
+	@mkdir -p $(GO_TOOLS_DIR)
+	@GOBIN=`pwd`/tools $(GOHOST) get github.com/golang/lint/golint
```

Comment:
I think what you want here is to set a temporary GOPATH, do the go get, copy the bin, and then remove the temporary go path. Otherwise you're still polluting the host with pkgs etc.